crates/net/network/src/session/mod.rs

```diff
@@ -170,9 +171,27 @@ impl SessionManager {
         self.status
     }
 
-    /// Returns the session hello message.
-    pub fn hello_message(&self) -> HelloMessageWithProtocols {
-        self.hello_message.clone()
+    /// Returns the session default hello message.
+    pub fn hello_message(&self) -> &HelloMessageWithProtocols {
+        &self.hello_message
+    }
+
+    // Update hello message with extra protocols to support in the session.
+    fn _update_hello_message(
+        &self,
+        extra_conns: &[Box<dyn DynConnectionHandler>],
+    ) -> HelloMessageWithProtocols {
+        let hello_message = self.hello_message.clone();
+        if extra_conns.is_empty() {
+            return hello_message
+        }
+
+        let mut builder = HelloMessageWithProtocols::builder_from(hello_message);
+        for handler in extra_conns {
+            builder = builder.protocol(handler.protocol());
+        }
+
+        builder.build()
     }
```

Comment:
I don't think we need this.

I prefer the solution included in #5559 :

https://github.com/paradigmxyz/reth/blob/a9e36ccb9d3cdb6bfef0a24071b0a709709780c1/crates/net/eth-wire/src/hello.rs#L74-L91

and 

https://github.com/paradigmxyz/reth/blob/a9e36ccb9d3cdb6bfef0a24071b0a709709780c1/crates/net/network/src/session/mod.rs#L930-L931

so internally depending on the direction we find the matching extra protocols, pass them  to the authenticate function install them into the hello message